docs/protocol-methods.rst

```diff
@@ -499,6 +581,172 @@ Unsubscribe from a script hash, preventing future notifications if its :ref:`sta
   Note that :const:`False` might be returned even for something subscribed to earlier,
   because the server can drop subscriptions in rare circumstances.
 
+blockchain.outpoint.subscribe
+=============================
+
+Subscribe to a transaction outpoint (TXO), to get notifications about its status.
+A status involves up to two transactions: the funding transaction that creates
+the TXO (as one of its outputs), and the spending transaction that uses it
+as an input (spends it).
+
+**Signature**
+
+  .. function:: blockchain.outpoint.subscribe(tx_hash, txout_idx, spk_hint=None)
```

Comment:
One thing that came up recently is performance issues around large transactions. Perhaps the call could instead accept **sorted** array of tuples `(tx_hash, txout_idx)`, `tx_hash` being more significant? That way implementations might be able to optimize database reads. As an alternative it could accept ranges of indexes.

Reply:
> One thing that came up recently is performance issues around large transactions

How is a tx being large relevant? Do you mean the scenario where a tx has lots of outputs and the client wants to subscribe to many of them? Why repeat `tx_hash` in that case?

Or do you simply mean that for the scenario where a client wants to subscribe to many outpoints (in general) it would be nice to allow batching the subscriptions? If so, what about `spk_hint`; shouldn't that be part of the tuple as well?

In either case, I am inclined to say it's not worth it to complicate the API for this.

Reply:
> Do you mean the scenario where a tx has lots of outputs and the client wants to subscribe to many of them?

Yes.

> Why repeat `tx_hash` in that case?

Good point, use array for indexes instead. Maybe an empty array could mean "all"?